docs/src/modules/spring/pages/actions-publishing-subscribing.adoc

```diff
@@ -1 +1,100 @@
-= Publishing and Subscribing
+= Publishing and Subscribing with Actions
+
+include::ROOT:partial$include.adoc[]
+
+A very common use case when building Microservices is to publish and subscribe to a stream of events. The source of events can be the journal of an event sourced entity, the value entity state changes, a https://cloud.google.com/pubsub/docs/overview[Google Cloud Pub/Sub] or Apache Kafka topic for asynchronous messaging between services.
+
+With Actions you can:
+
+- subscribe to events emitted by an event sourced entity within the same service.
+- subscribe to state changes emitted by a value entity within the same service.
+- subscribe to events from Event Sourced Entities published as xref:service-to-service.adoc[service to service eventing]
+- subscribe to external events from https://cloud.google.com/pubsub/docs/overview[Google Cloud Pub/Sub] or https://kafka.apache.org/[Apache Kafka].
+- publish events to a Google Cloud Pub/Sub or Apache Kafka topic.
+
+Messages are guaranteed to be delivered at least once. This means that receivers must be able to handle duplicate messages.
+
+== Subscribing to an Event Sourced Entity
+
+You can subscribe an Action to events from an Event Source Entity. To receive messages from the entity, you can annotate a method with `@Subscribe.EventSourcedEntity` and specify the class of the entity. The input type of the method must be the same of the events generated by the entity.
+
+The following example assumes the existence of an xref:eventsourced.adoc[Event Sourced Entity] counter that emits events of type `CounterEvent` with subtypes `ValueIncreased` and `ValueDecreased`.
+
+[source,java,indent=0]
+.src/main/java/com/example/actions/CounterJournalToTopicAction.java
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/CounterJournalToTopicAction.java[tag=sub-ESE-action]
+----
+<1> subscribing to the events from the Counter.
+<2> setting the method input type to the events produced by the counter.
+<3> any return is valid. See NOTE.
+
+The events from the entity are delivered to the Action. The implementation may vary, for this simplified example you are just logging it, but it could a forward to some other component or external service.
```

Comment:
```suggestion
The events from the entity are delivered to the Action and what to do with those is up for implementation. For this simplified example you are just logging it, but it could also do a forward to some other component or external service.
```